### PR TITLE
fix: incorrect level of envoy-gateway configmap vars

### PR DESCRIPTION
**Pull request by Xunzhuo**
fix: incorrect level of envoy-gateway configmap vars

Signed-off-by: bitliu <bitliu@tencent.com>

### CODE REVIEW

File: internal/envoygateway/config/decoder.go [hunk 1]
```diff
@@ -28,16 +29,16 @@ func Decode(cfgPath string) (*v1alpha1.EnvoyGateway, error) {
...
 		gvk.Kind != v1alpha1.KindEnvoyGateway {
-		return nil, err
+		return nil, errors.New("failed to decode unmatched resource types")
 	}
 
 	// Attempt to cast the object.
 	eg, ok := obj.(*v1alpha1.EnvoyGateway)
 	if !ok {
-		return nil, err
+		return nil, errors.New("failed to convert obj to EnvoyGateway types")
```
Comment (danehans) [inline internal/envoygateway/config/decoder.go:41]: s/obj/object/

s/types/type/

**Review by danehans — COMMENTED** [review]
A couple of small nits. Otherwise /lgtm.

File: internal/envoygateway/config/decoder.go (hunk 1 above)
Comment (Xunzhuo) [reply internal/envoygateway/config/decoder.go]: done.

**Review by danehans — COMMENTED** [review]
ONe last nit

File: internal/envoygateway/config/decoder.go [hunk 2]
```diff
@@ -28,16 +29,16 @@ func Decode(cfgPath string) (*v1alpha1.EnvoyGateway, error) {
 	}
 
 	// Figure out the resource type from the Group|Version|Kind.
-	if gvk.Group != v1alpha1.GroupVersion.Group &&
-		gvk.Version != v1alpha1.GroupVersion.Version &&
+	if gvk.Group != v1alpha1.GroupVersion.Group ||
+		gvk.Version != v1alpha1.GroupVersion.Version ||
 		gvk.Kind != v1alpha1.KindEnvoyGateway {
-		return nil, err
+		return nil, errors.New("failed to decode unmatched resource types")
```
[the remark this answers is not in the document]
Comment (Xunzhuo) [reply internal/envoygateway/config/decoder.go]: done. Thanks for reviews @danehans

**Review by danehans — COMMENTED** [review]
/lgtm

File: internal/envoygateway/config/decoder_test.go [hunk 3]
```diff
@@ -72,7 +153,7 @@ func TestDecode(t *testing.T) {
 				require.NoError(t, err)
 				require.Equal(t, tc.out, eg)
 			} else {
-				require.Error(t, err, "An error was expected")
+				require.Equal(t, (!reflect.DeepEqual(tc.out, eg) || err != nil), true)
```
Comment (arkodg) [inline internal/envoygateway/config/decoder_test.go:156]: this is hard to read :)

**Review by arkodg — APPROVED** [review]

**Pull request merged**